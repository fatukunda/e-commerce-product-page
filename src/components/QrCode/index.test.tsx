import { render, cleanup, fireEvent } from "@testing-library/react";
import renderer from "react-test-renderer";
import { Provider as ReduxProvider } from "react-redux";

import QRCode from ".";
import mockStore from "../mocks/store";

const renderQRCode = () =>
  render(
    <ReduxProvider store={mockStore()}>
      <QRCode url="https://test-url" />
    </ReduxProvider>
  );
describe("Tests for <QRCode/>", () => {
  afterEach(cleanup);
  test("QRCode: Snapshot", () => {
    const component = renderer.create(
      <ReduxProvider store={mockStore()}>
        <QRCode url="https://test-url" />
      </ReduxProvider>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("Renders without crashing", () => {
    renderQRCode();
  });
});
