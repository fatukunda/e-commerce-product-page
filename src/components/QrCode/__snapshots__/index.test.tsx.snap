// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for <QRCode/> QRCode: Snapshot 1`] = `
<div
  className="bg-gray-100 rounded-r-md mb-4 flex flex-col items-center justify-center"
>
  <canvas
    height={128}
    style={
      Object {
        "height": 128,
        "width": 128,
      }
    }
    width={128}
  />
</div>
`;
